examples/rex_planar_quadrotor_mpc.jl

```diff
@@ -104,73 +149,84 @@ function generate_planar_quadrotor_data()
+    X_train_mpc = Matrix{Vector{Float64}}(undef, N_tsim, num_train_mpc)
+    U_train_mpc = Matrix{Vector{Float64}}(undef, N_tsim-1, num_train_mpc)
 
     for i = 1:num_train_mpc
 
         x0 = initial_conditions_mpc_train[i]
-        X = nominal_trajectory(x0,N,dt)
-        U = [copy(Problems.trim_controls(model_real)) for k = 1:N]
-        T = range(0,t_sim,step=dt)
+        X = nominal_trajectory(x0,N_tf,dt)
+        U = [copy(Problems.trim_controls(model_real)) for k = 1:N_tf]
+        T = range(0,tf,step=dt)
 
         mpc = TrackingMPC(dmodel_nom, X, U, Vector(T), Qmpc, Rmpc, Qfmpc; Nt=Nt)
-        X_sim,U_sim,T_sim = simulatewithcontroller(dmodel_real, mpc, X[1], T[end], T[2])
+        X_sim,U_sim,T_sim = simulatewithcontroller(dmodel_real, mpc, X[1], t_sim, T[2])
             
         X_train_mpc[:,i] = X_sim
         U_train_mpc[:,i] = U_sim
     end
 
     # i = 5
-    # T_sim = range(0,t_sim,step=dt)
+    # T_ref = range(0,tf,step=dt)
     # X_ref = nominal_trajectory(initial_conditions_mpc_train[i],N,dt,final_conditions_mpc_train[i])
 
-    # plotstates(T_sim, X_ref, inds=1:3, xlabel="time (s)", ylabel="states",
+    # plotstates(T_ref, X_ref, inds=1:3, xlabel="time (s)", ylabel="states",
     #             label=["x (ref)" "y (ref)" "θ (ref)"], legend=:right, lw=2,
     #             linestyle=:dot, color=[1 2 3])
-    # plotstates!(T_sim, X_train_mpc[:, i], inds=1:3, xlabel="time (s)", ylabel="states",
+    # plotstates!(T_ref, X_train_mpc[:, i], inds=1:3, xlabel="time (s)", ylabel="states",
     #             label=["x (mpc)" "θ (mpc)"], legend=:right, lw=2,
     #             linestyle=:dot, color=[1 2 3])
 
     # Generate test data
 
-    X_test_real_mpc = Matrix{Vector{Float64}}(undef, N, num_train_mpc)
-    X_test_nom_mpc = Matrix{Vector{Float64}}(undef, N, num_train_mpc)
+    X_test_real_mpc = Matrix{Vector{Float64}}(undef, N_tf, num_test_mpc)
+    X_test_nom_mpc = Matrix{Vector{Float64}}(undef, N_tsim, num_test_mpc)
 
-    U_test_real_mpc = Matrix{Vector{Float64}}(undef, N-1, num_test_mpc)
-    U_test_nom_mpc = Matrix{Vector{Float64}}(undef, N-1, num_test_mpc)
+    U_test_real_mpc = Matrix{Vector{Float64}}(undef, N_tf-1, num_test_mpc)
+    U_test_nom_mpc = Matrix{Vector{Float64}}(undef, N_tsim-1, num_test_mpc)
 
     for i = 1:num_test_mpc
 
         x0 = initial_conditions_mpc_test[i]
-        X = nominal_trajectory(x0,N,dt)
-        U = [copy(Problems.trim_controls(model_real)) for k = 1:N]
-        T = range(0,t_sim,step=dt)
+        X = nominal_trajectory(x0,N_tf,dt)
+        U = [copy(Problems.trim_controls(model_real)) for k = 1:N_tf]
+        T = range(0,tf,step=dt)
         
         mpc_nom = TrackingMPC(dmodel_nom, X, U, Vector(T), Qmpc, Rmpc, Qfmpc; Nt=Nt)
-        X_nom,U_nom,T_nom = simulatewithcontroller(dmodel_real, mpc_nom, X[1], T[end], T[2])
+        X_nom,U_nom,T_nom = simulatewithcontroller(dmodel_real, mpc_nom, X[1], t_sim, T[2])
 
         X_test_real_mpc[:,i] = X
```

Comment:
this naming is pretty confusing. `X_test_real_mpc` is the reference trajectory and `X_test_nom_mpc` is the result of the simulation? How about `X_test_ref_mpc` and `X_test_mpc`?